backend/windmill-api/src/slack_approvals.rs

```diff
@@ -0,0 +1,1133 @@
+                                Error::BadRequest("No trigger_id found in payload.".to_string())
+                            })?;
+
+                            let value_str = action.value.as_ref().ok_or_else(|| {
+                                Error::BadRequest("No action value found".to_string())
+                            })?;
+
+                            let parsed_value: ModalActionValue = serde_json::from_str(value_str)
+                                .map_err(|_| {
+                                    Error::BadRequest("Invalid JSON in action value".to_string())
+                                })?;
+
+                            let w_id = &parsed_value.w_id;
+                            let path = &parsed_value.path;
+                            let approver = parsed_value.approver.as_deref();
+                            let message = parsed_value.message.as_deref();
+                            let job_id = Uuid::parse_str(&parsed_value.job_id)?;
+                            let flow_step_id = parsed_value.flow_step_id.as_deref();
+
+                            let slack_token = get_slack_token(&db, path, w_id).await?;
+                            let client = Client::new();
+                            let container = payload.container.ok_or_else(|| {
+                                Error::BadRequest("No container found.".to_string())
+                            })?;
+
+                            open_modal_with_blocks(
+                                &client,
+                                slack_token.as_str(),
+                                trigger_id,
+                                db,
+                                w_id,
+                                path,
+                                job_id,
+                                approver,
+                                message,
+                                flow_step_id,
+                                container,
+                            )
+                            .await
+                            .map_err(|e| {
+                                windmill_common::error::Error::BadRequest(e.to_string())
+                            })?;
+                        }
+                        _ => println!("Unknown action_id: {}", action.action_id),
+                    }
+                }
+            } else {
+                tracing::debug!("Unkown Slack Action!");
+            }
+        }
+    }
+
+    Ok(StatusCode::OK)
+}
+
+pub async fn request_slack_approval(
+    _authed: ApiAuthed,
+    Extension(db): Extension<DB>,
+    Path((w_id, job_id)): Path<(String, Uuid)>,
+    Query(approver): Query<QueryApprover>,
+    Query(message): Query<QueryMessage>,
+    Query(slack_resource_path): Query<QueryResourcePath>,
+    Query(channel_id): Query<QueryChannelId>,
+    Query(flow_step_id): Query<QueryFlowStepId>,
+) -> Result<StatusCode, Error> {
+    let slack_resource_path = match slack_resource_path.slack_resource_path {
```

Comment:
Consider making it non optional in the `Query<..>` type, same for others required ones